contents/df_end.md

```diff
@@ -141,36 +150,287 @@ filter(:name => ==("Alice"), grades_2020())
+However, this function is much more versatile than just removing columns, as we will descuss in this section.
+Lets create a dataset with multiple columns:
+
+```jl
+@sco responses()
+```
+
+Here, the data represents answers on five questions (`q1`, `q2`, ..., `q5`) in a questionnaire.
+Let's first "select" a few columns from this dataset.
+Again, we use symbols to specify columns:
+
+```jl
+s = "select(responses(), :id, :q1)"
+sco(s, process=without_caption_label)
+```
+
+But, we can also use strings if we want:
+
+```jl
+s = """select(responses(), "id", "q1", "q2")"""
+sco(s, process=without_caption_label)
+```
+
+To select everything _except_ one or more columns, use `Not` with a column name:
+
+```jl
+s = """select(responses(), Not(:q5))"""
+sco(s, process=without_caption_label)
+```
+
+Or, with multiple column names:
+
+```jl
+s = """select(responses(), Not([:q4, :q5]))"""
+sco(s, process=without_caption_label)
+```
+
+It's also fine to combine column names with `Not`:
+
+```jl
+s = """select(responses(), :q5, Not(:id))"""
+sco(s, process=without_caption_label)
+```
+
+Note how `q5` is now the first column.
+But, there is a more clever way to do this, and that is with `:`.
+The colon `:` can be thought of as "all the columns that we didn't include yet".
+For example:
+
+```jl
+s = """select(responses(), :q5, :)"""
+sco(s, process=without_caption_label)
+```
+
+Or, to put `q5` at the second position[^sudete]:
+
+[^sudete]: Thanks to Sudete on [Discourse](https://discourse.julialang.org/t/pull-dataframes-columns-to-the-front/60327/4) for this suggestion.
+
+```jl
+s = "select(responses(), 1, :q5, :)"
+sco(s, process=without_caption_label)
+```
+
+Even renaming is possible via select:
+
+```jl
+s = """select(responses(), 1 => "participant", :q1 => "age", :q2 => "nationality")"""
+sco(s, process=without_caption_label)
+```
+
+which, thanks to the `...` operator, we can also write as:
+
+```jl
+s = """
+    renames = (1 => "participant", :q1 => "age", :q2 => "nationality")
+    select(responses(), renames...)
+    """
+sco(s, process=without_caption_label)
+```
+
+> **Note:** Another example of splatting via the `...` operator is:
+> ```jl
+  s = """
+      V = ["a", "b", "c"]
+      joinpath(V...)
+      """
+  scob(s)
+  ```
+
+## Types and Missing Data {#sec:missing_data}
 
 ```{=comment}
 Try to combine with transformations
+
+categorical
+allowmissing
+disallowmissing
+```
+
+As discussed in @sec:load_save, `CSV.jl` will do its best to guess data types for your data.
+However, this won't always work perfectly.
+In this section, we show why good types are important and we fix wrong data types.
+To be more clear about the types, we show the text output for DataFrames instead of a pretty table.
+In this section, we work with the following dataset:
+
+```jl
+@sco process=string post=output_block wrong_types()
+```
+
+Because the date column has the wrong type, sorting won't work correctly:
```

Comment:
```suggestion
Because the `date` column has the wrong type, sorting won't work correctly:
```
I forgot our standard thing? Are we using ... for column names/variables names?